packages/terra-form-input/src/FormInput.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './FormInput.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The defaultValue of the input field. Use this to create an uncontrolled input.
+   */
+  defaultValue: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+  /**
+   * Whether the input is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Whether the input is invalid
+   */
+  isInvalid: PropTypes.bool,
+  /**
+   * Function to trigger when the input loses focus.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * Function to trigger when user changes the input value. Provide a function to create a controlled input.
+   */
+  onChange: PropTypes.func,
+  /**
+   * Function to trigger when the input gains focus.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Content to be displayed as the name.
+   */
+  name: PropTypes.string,
+  /**
+   * Whether the input is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the input field. Use this to create a controlled input.
+   */
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+};
+
+const defaultProps = {
+  defaultValue: undefined,
+  disabled: false,
+  isInvalid: false,
+  onBlur: undefined,
+  onChange: undefined,
+  onFocus: undefined,
+  name: null,
+  required: false,
+  value: undefined,
+};
+
+class FormInput extends React.Component {
+  render() {
+    const {
+      defaultValue,
+      disabled,
+      isInvalid,
+      onBlur,
+      onChange,
+      onFocus,
+      name,
+      required,
+      value,
+      ...customProps
+    } = this.props;
+
+    const attributes = Object.assign({}, customProps);
+    const formInputClassNames = cx([
```

Comment:
After you set formInputClassNames, delete the className key from the attributes array. Otherwise, you risk passing in an invalid class to the input.

Reply:
Doesn't [this](https://github.com/cerner/terra-core/blob/Form-Input-Element/packages/terra-form-input/src/FormInput.jsx#L109) ensure that the `className` prop always wins when prop collisions between any `className` that may be found in the `attributes` array that's spread on line 108?

Reply:
I just now saw that. Given that, I think that risk isn't present here.

Reply:
Cool :) it's always good to double check!